comp/core/tagger/taggerimpl/tagstore/entity_tags.go

```diff
@@ -6,18 +6,55 @@
 package tagstore
 
 import (
+	"maps"
 	"sort"
 	"strings"
+	"time"
 
 	"github.com/DataDog/datadog-agent/comp/core/tagger/taggerimpl/collectors"
 	"github.com/DataDog/datadog-agent/comp/core/tagger/types"
 	"github.com/DataDog/datadog-agent/pkg/tagset"
+	"github.com/DataDog/datadog-agent/pkg/util/flavor"
 )
 
-// EntityTags holds the tag information for a given entity. It is not
-// thread-safe, so should not be shared outside of the store. Usage inside the
-// store is safe since it relies on a global lock.
-type EntityTags struct {
+// This file defines the EntityTags interface which contains the tags for a
+// tagger entity.
+//
```

Comment:
should we still mention that it is not thread-safe and thus should not be shared outside the store? (Similar to the old comment)